README.md

```diff
@@ -1,2 +1,102 @@
+
+**Functional** tests should align with product ideas. If a user goes through a given flow, they should finish with some outcome. We've found these tests to be pretty difficult to work with as of late, and are trying to get rid of some of these.
+
+#### Organization
+
+In most cases, we should attempt to organize our file structure as follows:
+
+```
+repository/
+  lib/ (or something like it)
+  tests/
+    unit/
+    integration/
+    functional/
+```
+
+Additionally, the `package.json` should support the following commands:
+
+- `npm test`: runs a code linter and all three sets of tests
+- `npm unit`: runs the unit tests
+- `npm integration`: runs the integration tests
+- `npm functional`: runs the functional tests
+
+### Stubbing
+
+One of the most powerful tools we have (and are still learning to use well) is `sinon`. `sinon` is a stubbing and spying library that helps us assert that functions are called with what we expect.
+
+`sinon`'s [docs](http://sinonjs.org/docs/) are quite good, but we will talk about some guidelines to follow when using it:
+
+- In all tests where we are going to stub or spy on something, do so in a `before*`
+- In all tests where we do stub or spy on something,
+    - if it is on an object that is regenerated every test, no action is necessary
+    - if it is on a module or something persistent, always restore the function in an `after*`
+- Stub modules and functions in the correct scopes (meaning, as close to the test that uses it as possible, but hierarchal stubbing is fine)
+- When stubbing functions, we may stub the function to a 'truthy' behavior; doing so reduces code duplication, though the developer will need to adjust the stub(s) to test error cases
+- Collecting stubs into logical blocks and using the language to define what is being set up is encouraged (i.e. use `describe` blocks to collect and describe stubs, `it` blocks to say what happens)
+
+### Asserting
+
+Now that we have tests all squared away and written, let's talk about our assertions.
+
+First, a few options we have been using are `code` and `chai.assert`. [`code`](https://github.com/hapijs/code) is pretty straight forward, using the `expect` form of assertions. [`chai`](https://github.com/chaijs/chai)'s assertion library has worked well, especially with [promises](https://github.com/domenic/chai-as-promised), and we have been using chai's `assert` which helps keep straight which library with which we are working.
+
+We also have started using `sinon`'s assert features as well (e.g. `sinon.assert.calledWith`) which provides much nicer error messages than alternatives.
+
+A few notes to help guide assertions:
+
+- All tests must assert something
+- When testing with a stub in the same logical group, we must assert something about the stub
```

Comment:
take this slightly less literally. if you have a given block that uses some external thing to do something, stubbing it out and making sure it's used is one thing, and making sure that it might return the correct value or transform is another thing. breaking down tests into these really small parts can help a lot, and then don't require a ton of overhead to maintain.

short version: you only need _one_ test to assert something's being called. (This is also slightly related to when you should define stub behavior, but I'll discuss that elsewhere.)

if it accepts various inputs (as your second comment seemed to imply), you can always change what the stub returns in your test (but one may find issue with your code where you can pass different objects - that's beside the point)

examples are always good in this case (and we may have to put one into the README)

```javascript
// this is describing an 'enter' function on a 'user'. 'enter' sets a value 'seat'
// on the user, which it gets from Bar.prototype.stool
describe('enter function', function () {
  before(function () {
    sinon.stub(Bar.prototype, 'stool').returns(7) // some 'success' case
  })
  after(function () {
    Bar.prototype.stool.restore() // restore the stub
  })

  it('should use Bar to get an available stool', function () {
    user.enter()

    sinon.assert.calledOnce(Bar.prototype.stool)
    // called w/ some value `1`, internal to user
    sinon.assert.calledWithExactly(Bar.prototype.stool, 1)
  })

  it('should have the seat assigned to it', function () {
    user.enter()

    assert.equal(user.seat, 7, 'user has a seat')
  })

  it('should throw if no seats available', function () {
    Bar.prototype.stool.returns(-1)

    function _test () { user.enter() }
    expect(_test).to.throw(Error, 'no seat')
  })
})
```